### ISSUE TITLE
running test from sbt on zio-tests mis-reports the test (RC5)

### ISSUE THREAD

**Issue by TimPigden** [zio/zio#6589]
Sample output:
```
[info]     - continue until new solution cost > initial cost (test1) - test parsing criteria
[info]       Exception in thread "zio-fiber-971" optrak.executive.solverlog.TrackedEvent$TrackingError$LoggingError: LoggingError(cache mySolution expected cache item 0 but not found
[info]       deltaCache mySolution
...
[info]       	at optrak.solver.dsl.TestLoadedSolvers.spec(TestLoadedSolvers.scala:24)
[info] Done
[error] Error during tests:
[error] 	optrak.executive.solverlog.logfiles.TestSysInfo
[error] (Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 10 s, completed 9 Apr 2022, 13:53:27
```
note that the failing test is TestLoadedSolvers yet the Error during tests: 
is reporting an error in TestSysInfo - which in fact passed

**Comment by adamgfraser** [comment]
@TimPigden Thanks for reporting! Will take a look.

**Comment by TimPigden** [comment]
[wrong-test-error.zip](https://github.com/zio/zio/files/8457055/wrong-test-error.zip)
this illustrates problem with reporting wrong error.
Run test in sbt and see what errors get produced

**Comment by TimPigden** [comment]
[optimisation.zip](https://github.com/zio/zio/files/8457079/optimisation.zip)
when i run the test in sbt it reports the error in the wrong test

**Comment by adamgfraser** [comment]
Resolved by #6681.
